### ISSUE TITLE
git diff

### ISSUE THREAD

**Issue by dalduba** [isomorphic-git/isomorphic-git#1754]
Is there a chance that "git diff" will be implemented?

**Comment by dalduba** [comment]
The case is:
Edit some heavy data in a browser, send only diff to the server when saving.

**Comment by jcubic** [comment]
No, it will never be implemented because it requires a Diff library and it's out of scope. You can use Diff library of your choice (like jsDiff) and implement git diff yourself.

Check my project where I've added git diff without any issues using existing API, even before version 1.0

https://github.com/jcubic/git/blob/master/js/main.js#L842

**Comment by dalduba** [comment]
> No, it will never be implemented because it requires a Diff library and it's out of scope. You can use Diff library of your choice (like jsDiff) and implement git diff yourself.
> 
> Check my project where I've added git diff without any issues using existing API, even before version 1.0
> 
> https://github.com/jcubic/git/blob/master/js/main.js#L842

Thanks for your reply. Could you suggest a library for "git apply" implementation on js?

**Comment by dalduba** [comment]
Found, that jsdiff has Api for applying patches too.

But seems that jsDiff works only with text. May be  you know solutions for binary data?

**Comment by jcubic** [comment]
I don't think that you can diff binary data in canonical git

**Comment by dalduba** [comment]
"--binary" key enables binary patches generation 
https://superuser.com/a/1567733

**Comment by jcubic** [comment]
You will need to write your own library in JavaScript to create binary patches. But the git source code it Open Source so you can look at how it was implemented and translate that to JavaScript.
